core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -140,13 +143,19 @@ public NullWritable getCurrentKey() throws IOException, InterruptedException {
     protected abstract StructuredRecord getCurrentValue(StructuredRecord.Builder recordBuilder)
       throws IOException, InterruptedException;
 
+    protected abstract StructuredRecord getCurrentValueNull() throws IOException, InterruptedException;
```

Comment:
this is a strange contract. The fact that you return null from one of the implementations and expect it never to be used by the parent means it's not a clean abstraction.